mlflow/entities/span.py

```diff
@@ -308,6 +329,41 @@ def set_outputs(self, outputs: Any):
         """Set the output values to the span."""
         self.set_attribute(SpanAttributeKey.OUTPUTS, outputs)
 
+    def set_chat_messages(
```

Comment:
With the same reason above, can we move this out to separate utility? We don't want to have a type-specific logic within the general Span class.

Reply:
makes sense, what do you think about some fluent api like `mlflow.set_span_chat_messages(span, messages)`?

Reply:
Yea the API sounds good, I'd put it in `mlflow.tracing.xyz` not to overload `mlflow` namespace tho

Reply:
Another option is to put it into the model side:
```
mlflow.set_attributes(messages.to_span_attribute())
```
Probably the message model is less crowded than span model (cuz it is purpose built)

Reply:
i guess there's no harm, anyway probably most users will be using autologging so it shouldn't add too much friction